src/serialization/sb2.js

```diff
@@ -866,32 +867,35 @@ const parseBlock = function (sb2block, addBroadcastMsg, getVariableId, extension
                     // Single block occupies the input.
                     const parsedBlockDesc = parseBlock(providedArg, addBroadcastMsg, getVariableId, extensions,
                         parseState, comments, commentIndex);
-                    innerBlocks = [parsedBlockDesc[0]];
+                    innerBlocks = parsedBlockDesc[0] ? [parsedBlockDesc[0]] : [];
                     // Update commentIndex
                     commentIndex = parsedBlockDesc[1];
                 }
                 parseState.expectedArg = parentExpectedArg;
-                // Check if innerBlocks is an empty list.
-                // This indicates that all the inner blocks from the sb2 have
+
+                // Obscures any shadow.
+                shadowObscured = true;
```

Comment:
You should move this line below the `if (innerBlocks.length > 0)` block below since it's not being used till later